### PR TITLE
Add new function: aclObjectGetGroups

### PR DESCRIPTION
**Pull request by TracerDS**
This function will return all groups of an object.
Syntax: `table aclObjectGetGroups ( string theObject )`

Function returns `table` on success, `false` if something went wrong

### CODE REVIEW

**Comment by botder** [comment]
Please use the new argument parser.

**Comment by TracerDS** [comment]
> Please use the new argument parser.

Should I return a vector as a table then?

File: Server/mods/deathmatch/logic/luadefs/CLuaACLDefs.cpp [hunk 1]
```diff
@@ -1049,3 +1051,36 @@ int CLuaACLDefs::OOP_isObjectInACLGroup(lua_State* luaVM)
     lua_pushboolean(luaVM, false);
     return 1;
 }
+
+std::variant<std::vector<CAccessControlListGroup*>, bool> CLuaACLDefs::aclObjectGetGroups(SString strObject)
```
Comment (CrosRoad95) [inline Server/mods/deathmatch/logic/luadefs/CLuaACLDefs.cpp]: change type from `SString` to `std::string`, convert it later to "SString" if needed

File: Server/mods/deathmatch/logic/luadefs/CLuaACLDefs.cpp [hunk 2]
```diff
@@ -1049,3 +1051,37 @@ int CLuaACLDefs::OOP_isObjectInACLGroup(lua_State* luaVM)
...
+        objectType = CAccessControlListGroupObject::OBJECT_TYPE_RESOURCE;
+    }
+    else if (StringBeginsWith(szObjectAfterDot, "user."))
+    {
+        szObjectAfterDot += 5;
+        objectType = CAccessControlListGroupObject::OBJECT_TYPE_USER;
+    }
+    else
+    {
+        return false;
```
Comment (botder) [inline Server/mods/deathmatch/logic/luadefs/CLuaACLDefs.cpp]: This function should never anything else other than a list of groups. Throw an exception here.

File: Server/mods/deathmatch/logic/luadefs/CLuaACLDefs.h [hunk 3]
```diff
@@ -48,4 +48,7 @@ class CLuaACLDefs : public CLuaDefs
     LUA_DECLARE(aclGroupRemoveObject);
     LUA_DECLARE_OOP(isObjectInACLGroup);
     LUA_DECLARE(hasObjectPermissionTo);
+
+    static std::variant<std::vector<CAccessControlListGroup*>, bool>
+        aclObjectGetGroups(std::string strObject);
```
Comment (botder) [inline Server/mods/deathmatch/logic/luadefs/CLuaACLDefs.h]: This function returns a vector on every return path now, get rid of the variant.

**Review by botder — APPROVED** [review]

**Pull request merged**